zbot_unit_tests/test_03.py

```diff
@@ -1,28 +1,136 @@
-"""Runs Mujoco and real robot in parallel, for tuning the Z-Bot simulation parameters."""
+"""
+IMPORTANT: Before running this test, start the simulator server with:
+    kos-sim zbot-v2
+"""
 
 import asyncio
-import logging
+import sys
+import subprocess
+import time
+from pykos import KOS
 
-import colorlogging
-import pykos
+ACTUATOR_MAPPING = {
+    # Arms
+    "left_shoulder_yaw": 11,
+    "left_shoulder_pitch": 12,
+    "left_elbow": 13,
+    "left_wrist": 14,
+    "right_shoulder_yaw": 21,
+    "right_shoulder_pitch": 22,
+    "right_elbow": 23,
+    "right_wrist": 24,
+    # Legs (for zeroing)
+    "left_hip_roll": 31,
+    "left_hip_yaw": 32,
+    "left_hip_pitch": 33,
+    "left_knee": 34,
+    "left_ankle": 35,
+    "right_hip_roll": 41,
+    "right_hip_yaw": 42,
+    "right_hip_pitch": 43,
+    "right_knee": 44,
+    "right_ankle": 45,
+}
 
-logger = logging.getLogger(__name__)
 
+async def run_robot(kos: KOS, is_real: bool) -> None:
+    # Configure actuators
+    for actuator_id in ACTUATOR_MAPPING.values():
+        if is_real:
+            kp, kd = 32, 32  # Real robot uses 32, 32 for all joints
+        else:
+            # Simulation gains (higher for better tracking)
+            gains = {
+                # Arm gains
+                11: (50, 5),
+                12: (50, 5),
+                13: (50, 5),
+                14: (30, 3),
+                21: (50, 5),
+                22: (50, 5),
+                23: (50, 5),
+                24: (30, 3),
+                # Leg gains
+                31: (150, 15),
+                32: (150, 15),
+                33: (150, 15),
+                34: (150, 15),
+                35: (100, 10),
+                41: (150, 15),
+                42: (150, 15),
+                43: (150, 15),
+                44: (150, 15),
+                45: (100, 10),
+            }
```

Comment:
interesting. so basically we need to adjust these values until they match the real robot very carefully